content/blog/2016/2016-05-10-jenkins-20-vjam.adoc

```diff
@@ -0,0 +1,355 @@
+____
+
+Yes, via the `env` variables provided in the DSL scope. Using an `if` statement,
+one could guard specific behaviors with:
+
+[source,groovy]
+----
+if (env.CHANGE_ID != null) {
+    /* do things! */
+}
+----
+
+____
+Let's say I'm building RPMs with Jenkins and use build number as an RPM
+version/release number. Is there a way to maintain build numbers and leverage
+versioning of `Jenkinsfile`?
+____
+
+
+Through the `env` variable, it's possible to utilize `env.BUILD_NUMBER` or the
+SCM commit ID, etc.
+
+____
+Love the snippet generator! Any chance of separating it out from the pipeline
+into a separate page on its own, available in the left nav?
+____
+
+Yes, this is tracked in
+link:https://issues.jenkins-ci.org/browse/JENKINS-31831[JENKINS-31831]
+
+____
+Any tips on pre-creating the admin user credential and selecting plugins to
+automate the Jenkins install?
+____
+
+There are various link:https://github.com/jenkinsci/puppet-jenkins[configuration
+management modules] which provide parts of this functionality.
+
+
+
+____
+I'm looking at the pipeline syntax (in Jenkins 2.0) how do I detect a
+`step([...])` has failed and create a notification inside the `Jenkinsfile`?
+____
+
+This can be done by wrapping a step invocation with a Groovy `try/catch` block.
+See also link:https://issues.jenkins-ci.org/browse/JENKINS-28119[JENKINS-28119]
+
+
+==== User Interface/Experience
+____
+Is the user experience same as before when we replace the Jenkins.war(1.x to
+2.x) in an existing (with security in place) installation?
+____
+
+You will get the new UI features like redesigned configuration forms, but the
+initial setup wizard will be skipped. In its stead, Jenkins will offer to
+install Pipeline-related functionality.
+
+
+____
+Is it possible to use custom defined syntax highlighting ?
+____
+
+Within the Pipeline script editor itself, no. It is using the ACE editor system,
+so it may be possible for a plugin to change the color scheme used.
+
+
+____
+Can you elaborate on what the Blue Ocean UI is? Is there a link or more
+information on it?
+____
+
+Blue Ocean is the name of user experience an design project, unfortunately at
+this point in time there is not more information available on it.
+
+
+==== General
+____
+How well this integrate with cloud environment?
+____
+
+The Jenkins master and agents can run easily in any public cloud environment
+that supports running Java applications. Through the
+link:https://wiki.jenkins-ci.org/display/JENKINS/Amazon+EC2+Plugin[EC2],
+link:https://wiki.jenkins-ci.org/display/JENKINS/JClouds+Plugin[JClouds] or
+link:https://wiki.jenkins-ci.org/display/JENKINS/Azure+Slave+Plugin[Azure]
+plugins it is possible to dynamically provision new build agents on a configured
+cloud provider.
+
+____
+Are help texts and other labels and messages updated for other localizations /
+languages as well?
+____
+
+Practically every string in Jenkins is localizable. The extent to which those
```

Comment:
I would disagree. I'd prefer something like "Practically almost every string in Jenkins WebUI can be made localizable. The most of the plugins support it, but it depends on the implementation..."